app/models/project_services/composer_service.rb

```diff
@@ -0,0 +1,474 @@
+              ['Shopware Theme', 'shopware-theme'],
+              ['SilverStripe Module', 'silverstripe-module'],
+              ['SilverStripe Theme', 'silverstripe-theme'],
+              ['SMF Module', 'smf-module'],
+              ['SMF Theme', 'smf-theme'],
+              ['Symfony1 Plugin', 'symfony1-plugin'],
+              ['Tusk Asset', 'tusk-asset'],
+              ['Tusk Command', 'tusk-command'],
+              ['Tusk Task', 'tusk-task'],
+              ['TYPO3 CMS Extension', 'typo3-cms-extension'],
+              ['TYPO3 Flow Boilerplate', 'typo3-flow-boilerplate'],
+              ['TYPO3 Flow Build', 'typo3-flow-build'],
+              ['TYPO3 Flow Framework', 'typo3-flow-framework'],
+              ['TYPO3 Flow Package', 'typo3-flow-package'],
+              ['TYPO3 Flow Plugin', 'typo3-flow-plugin'],
+              ['TYPO3 Flow Site', 'typo3-flow-site'],
+              ['Wolf CMS Plugin', 'wolfcms-plugin'],
+              ['WordPress Core', 'wordpress-core'],
+              ['WordPress Must Use Plugin', 'wordpress-muplugin'],
+              ['WordPress Plugin', 'wordpress-plugin'],
+              ['WordPress Theme', 'wordpress-theme'],
+              ['Zend Extra', 'zend-extra'],
+              ['Zend Library', 'zend-library'],
+              ['Zend Module', 'zend-module'],
+              ['Zikula Module', 'zikula-module'],
+              ['Zikula Theme', 'zikula-theme']
+            ],
+            default_choice: 'library'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Branches:', fields:
+        [
+          { type: 'checkbox',
+            name: 'export_branches',
+            title: 'Export',
+          },
+          { type: 'text',
+            name: 'branch_filters',
+            title: 'Filters',
+            placeholder: 'branches you wish to export comma separated.',
+            hint: 'Separate branches with commas. '\
+                  'Leave blank to export all branches.'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Tags:', fields:
+        [
+          { type: 'checkbox',
+            name: 'export_tags',
+            title: 'Export'
+          },
+          { type: 'text',
+            name: 'tag_filters',
+            title: 'Filters',
+            placeholder: 'tags you wish to export comma separated.',
+            hint: 'Separate tags with commas. '\
+                  'Leave blank to export all tags.'
+          }
+        ]
+      },
+      { type: 'fieldset', legend: 'Advanced:', fields:
+        [
+          { type: 'textarea',
+            name: 'custom_json',
+            title: 'Custom JSON',
+            placeholder: 'custom json to use for exporting this package.'
+          }
+        ]
+      }
+    ]
+  end
+
+  # disable test button
+  def can_test?
+    false
+  end
+
+  def supported_events
+    %w(push tag_push)
+  end
+
+  def execute(push_data)
+    process_project
+  end
+
+  def process_project
```

Comment:
Please move this to a separate class in lib. The service is way too heavy this way.